compiler/rustc_codegen_cranelift/src/constant.rs

```diff
@@ -471,6 +474,8 @@ fn define_all_allocs(tcx: TyCtxt<'_>, module: &mut dyn Module, cx: &mut Constant
                         .principal()
                         .map(|principal| tcx.instantiate_bound_regions_with_erased(principal)),
                 ),
+                // TODO
+                GlobalAlloc::Type { .. } => todo!(),
```

Comment:
This probably needs something like `data_id_for_type` similar to `data_id_for_vtable`, which can then also be used in `codegen_const_value`.